pandas/core/internals.py

```diff
@@ -1528,7 +1552,9 @@ def _try_coerce_result(self, result):
                 result = result.astype('m8[ns]')
             result[mask] = tslib.iNaT
         elif isinstance(result, np.integer):
-            result = lib.Timedelta(result)
+            result = self._box_func(result)
+        elif convert_float and isinstance(result, np.float):
```

Comment:
This is needed to coerce ``.quantile`` result. Any problem if we coerce ``float`` to ``datetime/timedelta`` always?